torch/csrc/distributed/autograd/rpc_messages/rpc_with_profiling_req.cpp

```diff
@@ -0,0 +1,151 @@
+#include <torch/csrc/distributed/autograd/rpc_messages/rpc_with_profiling_req.h>
+#include <torch/csrc/distributed/rpc/utils.h>
+#include <torch/csrc/jit/serialization/pickle.h>
+#include <vector>
+
+namespace torch {
+namespace distributed {
+namespace autograd {
+
+using rpc::RpcCommandBase;
+
+// This constructor is called when creating the RpcWithProfilingReq on the
+// client.
+RpcWithProfilingReq::RpcWithProfilingReq(
+    rpc::worker_id_t fromWorkerId,
+    rpc::MessageType messageType,
+    rpc::Message&& wrappedMessage,
+    const torch::autograd::profiler::ProfilerConfig& profilerConfig)
+    : fromWorkerId_(fromWorkerId),
+      messageType_(messageType),
+      wrappedMessage_(std::move(wrappedMessage)),
+      profilerConfig_(profilerConfig) {
+  tensors_ = wrappedMessage_.tensors();
+  TORCH_INTERNAL_ASSERT(
+      messageType_ == rpc::MessageType::RUN_WITH_PROFILING_REQ,
+      "Incorrect message type");
+  wrappedMessageType_ = wrappedMessage_.type();
+}
+
+// this constructor is only called in fromMessage() which is called in
+// deserializeRequest(). It is called when reconstructing the
+// RpcWithProfilingReq on the remote end.
+RpcWithProfilingReq::RpcWithProfilingReq(
+    rpc::worker_id_t fromWorkerId,
+    rpc::MessageType messageType,
+    std::unique_ptr<rpc::RpcCommandBase> wrappedRpc,
+    rpc::MessageType wrappedMessageType,
+    std::vector<torch::Tensor> tensors,
+    torch::autograd::profiler::ProfilerConfig& profilerConfig)
+    : fromWorkerId_(fromWorkerId),
+      messageType_(messageType),
+      wrappedRpc_(std::move(wrappedRpc)),
+      wrappedMessageType_(wrappedMessageType),
+      tensors_(std::move(tensors)),
```

Comment:
Looks like `wrappedMessage_` is not set in this case? Do we still need to make `wrappedMessage_` a member field, or does it make sense if we just keep the `tensors_`, `payload_`, etc. of it?

Reply:
It is set in the constructor above this, and we use the id, type, payload, and tensors field of it in that case (in particular, we use that constructor when making this message on the client in preparation for send). I think it's better to keep it in the wrappedMessage_ field instead of adding a bunch of args?